### PR TITLE
Throw LogicException on Windows

### PR DESCRIPTION
**Pull request by WyriHaximus**
Since pipes Windows always block this PR introduces always throwing an exception so we don't allocate unnecessary resources.

Follows up #41

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -309,9 +309,10 @@ of the actual exit code.
 
 ### Windows Compatibility
 
-Due to the blocking nature of `STDIN`/`STDOUT`/`STDERR` pipes on Windows we we can 
-not guarantee this package works as expected on Windows directly. However this package 
-does work on [`Windows Subsystem for Linux`](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux) 
+Due to the blocking nature of `STDIN`/`STDOUT`/`STDERR` pipes on Windows we can 
+not guarantee this package works as expected on Windows directly. As such when 
+instantiating `Process` it throws an exception when on Windows directly.
```
Comment (jsor) [inline README.md]: The doubling of "directly" at the end of the 2 sentences could probably be avoided.

File: src/Process.php [hunk 2]
```diff
@@ -45,10 +45,15 @@ class Process extends EventEmitter
     * @param string $cwd     Current working directory or null to inherit
     * @param array  $env     Environment variables or null to inherit
     * @param array  $options Options for proc_open()
+    * @throws LogicException On windows
     * @throws RuntimeException When proc_open() is not installed
     */
     public function __construct($cmd, $cwd = null, array $env = null, array $options = array())
     {
+        if (substr(strtolower(PHP_OS), 0, 3) === 'win') {
+            throw new \LogicException('Windows isn\'t supported.');
```
Comment (jsor) [inline src/Process.php]: The exception message could be a little bit more descriptiove, something like "Windows isn't supported due to the blocking nature of STDIN/STDOUT/STDERR pipes."

**Comment by WyriHaximus** [comment]
@jsor updated the PR with your suggestions :+1:

File: src/Process.php [hunk 3]
```diff
@@ -45,10 +45,15 @@ class Process extends EventEmitter
     * @param string $cwd     Current working directory or null to inherit
     * @param array  $env     Environment variables or null to inherit
     * @param array  $options Options for proc_open()
+    * @throws LogicException On windows
     * @throws RuntimeException When proc_open() is not installed
     */
     public function __construct($cmd, $cwd = null, array $env = null, array $options = array())
     {
+        if (substr(strtolower(PHP_OS), 0, 3) === 'win') {
+            throw new \LogicException('Windows isn\'t supported due to the blocking nature of STDIN/STDOUT/STDERR pipes.');
```
Comment (mdrost) [inline src/Process.php]: Requiring `not windows` is same kind of precondition as requiring `proc_open` function to exists so both exceptions should be same type.
IMHO both should be LogicException.

Comment (WyriHaximus) [inline src/Process.php]: Makes sense to me, what do you think @clue and @jsor? IMHO it should be done in a different PR though

Comment (clue) [inline src/Process.php]: No strong opinion here, but makes sense :+1:

Comment (WyriHaximus) [inline src/Process.php]: @mdrost care to open a PR for it?

Comment (mdrost) [inline src/Process.php]: @WyriHaximus https://github.com/reactphp/child-process/pull/49

Comment (WyriHaximus) [inline src/Process.php]: @mdrost Yup saw it :+1:
